src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,28 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return ((this.length == rectangle.length) || (this.length != null
+                && this.length.equals(rectangle.length)))
+                && ((this.width == rectangle.width) || (this.width != null
+                && this.width.equals(rectangle.width)))
+                && ((this.color == rectangle.color) || (this.color != null
+                && this.color.equals(rectangle.color)));
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * length.hashCode() * width.hashCode() * color.hashCode();
```

Comment:
What will happen if one of field, for example length or width == 0?

Reply:
more collisions would happen